ts/screens/onboarding/OnboardingServicesPreferenceScreen.tsx

```diff
@@ -19,29 +19,80 @@ import { IOStyles } from "../../components/core/variables/IOStyles";
 import { navigateToOnboardingServicePreferenceCompleteAction } from "../../store/actions/navigation";
 import { useManualConfigBottomSheet } from "../profile/components/services/ManualConfigBottomSheet";
 import { ServicesPreferencesModeEnum } from "../../../definitions/backend/ServicesPreferencesMode";
+import {
+  isServicesPreferenceModeSet,
+  profileSelector,
+  profileServicePreferencesModeSelector
+} from "../../store/reducers/profile";
+import { profileUpsert } from "../../store/actions/profile";
+import { withLoadingSpinner } from "../../components/helpers/withLoadingSpinner";
+import { showToast } from "../../utils/showToast";
 
-
+type NavigationProps = {
+  isFirstOnboarding: boolean;
+};
 type Props = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps> &
-  NavigationInjectedProps<{ isOldUser?: true }>;
+  NavigationInjectedProps<NavigationProps>;
 
 const OnboardingServicesPreferenceScreen = (
   props: Props
 ): React.ReactElement => {
-  const { present: confirmManualConfig } = useManualConfigBottomSheet();
+  const isFirstOnboarding = props.navigation.getParam("isFirstOnboarding");
+  // if the user is not new and he/she hasn't a preference set, pre-set with AUTO mode
+  const mode =
+    !isFirstOnboarding &&
+    props.profileServicePreferenceMode === ServicesPreferencesModeEnum.LEGACY
+      ? ServicesPreferencesModeEnum.AUTO
+      : props.profileServicePreferenceMode;
+  const [modeSelected, setModeSelected] = React.useState<
+    ServicesPreferencesModeEnum | undefined
+  >(mode);
+  const [prevPotProfile, setPrevPotProfile] = React.useState<
+    typeof props.potProfile
+  >(props.potProfile);
+  React.useEffect(() => {
+    // when the user made a choice (the profile is right updated), navigate to thank-you page
+    if (isServicesPreferenceModeSet(props.profileServicePreferenceMode)) {
+      props.onContinue();
+      return;
+    }
+    // show error toast only when the profile updating fails
+    // otherwise, if the profile is in error state, the toast will be shown immediately without any updates
+    if (!pot.isError(prevPotProfile) && pot.isError(props.potProfile)) {
+      showToast(I18n.t("global.genericError"));
+    }
+    setPrevPotProfile(props.potProfile);
+  }, [props.potProfile]);
 
-  const onConfirmAction = (mode: ServicesPreferencesModeEnum) => {
+  const handleOnContinue = () => {
+    if (modeSelected) {
+      props.onServicePreferenceSelected(modeSelected);
+    }
+  };
+  const { present: confirmManualConfig } = useManualConfigBottomSheet();
+  const handleOnSelectMode = (mode: ServicesPreferencesModeEnum) => {
+    // if user's choice is 'manual', open bottom sheet to ask confirmation
     if (mode === ServicesPreferencesModeEnum.MANUAL) {
-      return confirmManualConfig(props.onContinue);
+      void confirmManualConfig(() =>
+        setModeSelected(ServicesPreferencesModeEnum.MANUAL)
```

Comment:
I guess confirm button of the bottom sheet should act as the confirm CTA of the screen, could this solve?
```suggestion
        props.onServicePreferenceSelected(ServicesPreferencesModeEnum.MANUAL)
```

Reply:
I really don't know that is the expected behavior.
Your suggestion solves! thanks
 f99aa625142331466501f218960ec412d39f01df